chain/indexer.go

```diff
@@ -57,18 +60,19 @@ var _ TipSetObserver = (*TipSetIndexer)(nil)
 
 // A TipSetWatcher waits for tipsets and persists their block data into a database.
 type TipSetIndexer struct {
-	window            time.Duration
-	storage           model.Storage
-	processors        map[string]TipSetProcessor
-	messageProcessors map[string]MessageProcessor
-	actorProcessors   map[string]ActorProcessor
-	name              string
-	persistSlot       chan struct{} // filled with a token when a goroutine is persisting data
-	lastTipSet        *types.TipSet
-	node              lens.API
-	opener            lens.APIOpener
-	closer            lens.APICloser
-	addressFilter     *AddressFilter
+	window                     time.Duration
+	storage                    model.Storage
+	processors                 map[string]TipSetProcessor
+	messageProcessors          map[string]MessageProcessor
+	messageExecutionProcessors map[string]MessageExecutionProcessor
```

Comment:
the new processor for message executions, unable to reuse the existing one since lily is required for this workflow.